### PR TITLE
Mastodon New Sources & Actions

### LINKED ISSUE #5019: Mastodon
**Is there a specific app this action is for?**
Mastodon

**Actions**
- [Post a new status](https://docs.joinmastodon.org/methods/statuses/#create)
- [View a single status](https://docs.joinmastodon.org/methods/statuses/#get)
- [Delete a status](https://docs.joinmastodon.org/methods/statuses/#delete)
- [Favourite a status](https://docs.joinmastodon.org/methods/statuses/#favourite)
- [Undo favourite of a status](https://docs.joinmastodon.org/methods/statuses/#unfavourite)
- [Boost a status](https://docs.joinmastodon.org/methods/statuses/#boost)
- [Undo boost of a status](https://docs.joinmastodon.org/methods/statuses/#unreblog)
- [Bookmark a status](https://docs.joinmastodon.org/methods/statuses/#bookmark)
- [Undo bookmark of a status](https://docs.joinmastodon.org/methods/statuses/#unbookmark)
- [Mute a conversation](https://docs.joinmastodon.org/methods/statuses/#mute)
- [Unmute a conversation](https://docs.joinmastodon.org/methods/statuses/#unmute)
- [Pin status to profile](https://docs.joinmastodon.org/methods/statuses/#pin)
- [Unpin status from profile](https://docs.joinmastodon.org/methods/statuses/#unpin)
- [Edit a status](https://docs.joinmastodon.org/methods/statuses/#edit)
- [View public timeline](https://docs.joinmastodon.org/methods/timelines/#public)
- [View hashtag timeline](https://docs.joinmastodon.org/methods/timelines/#tag)
- [View home timeline](https://docs.joinmastodon.org/methods/timelines/#home)
- [View list timeline](https://docs.joinmastodon.org/methods/timelines/#list)
- [Perform a search](https://docs.joinmastodon.org/methods/search/#v2)

**Triggers**
- [New status](https://docs.joinmastodon.org/methods/statuses/#create)
- [New status boosted](https://docs.joinmastodon.org/methods/statuses/#reblogged_by)
- [New status favourited](https://docs.joinmastodon.org/methods/statuses/#favourited_by)

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #5019

### CODE REVIEW

File: components/mastodon/actions/perform-search/perform-search.mjs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    q: {
+      type: "string",
+      label: "Query",
+      description: "The search query",
+    },
+    type: {
+      type: "string",
+      label: "Type",
+      description: "Specify whether to search for only `accounts`, `hashtags`, `statuses`",
+      options: [
```
Comment (alysonturing) [inline components/mastodon/actions/perform-search/perform-search.mjs:20]: you could move that options to a constant file

File: components/mastodon/actions/post-status/post-status.mjs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      type: "string",
+      label: "Spoiler Text",
+      description: "Text to be shown as a warning or subject before the actual content. Statuses are generally collapsed behind this field",
+      optional: true,
+    },
+    visibility: {
+      type: "string",
+      label: "Visibility",
+      description: "Sets the visibility of the posted status to `public`, `unlisted`, `private`, or `direct`",
+      options: [
```
Comment (alysonturing) [inline components/mastodon/actions/post-status/post-status.mjs:39]: you could move this to a constant file

File: components/mastodon/sources/new-status-boosted/new-status-boosted.mjs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import common from "../common/base.mjs";
+
+export default {
+  ...common,
+  key: "mastodon-new-status-boosted",
+  name: "New Status Boosted",
+  description: "Emit new event when the specified status is boosted",
```
Comment (alysonturing) [inline components/mastodon/sources/new-status-boosted/new-status-boosted.mjs:7]: you could also add link to the docs of the currently used endpoint on this source

File: components/mastodon/sources/new-status-favorited/new-status-favorited.mjs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import common from "../common/base.mjs";
+
+export default {
+  ...common,
+  key: "mastodon-new-status-favorited",
+  name: "New Status Favorited",
+  description: "Emit new event when the specified status is favorited",
```
Comment (alysonturing) [inline components/mastodon/sources/new-status-favorited/new-status-favorited.mjs:7]: you could also add link to the docs of the currently used endpoint on this source

File: components/mastodon/sources/new-status/new-status.mjs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import common from "../common/base.mjs";
+
+export default {
+  ...common,
+  key: "mastodon-new-status",
+  name: "New Status",
+  description: "Emit new event when a new status is posted to your Profile",
```
Comment (alysonturing) [inline components/mastodon/sources/new-status/new-status.mjs:7]: you could also add link to the docs of the currently used endpoint on this source

**Review by alysonturing — DISMISSED** [review]
LGTM! 
I'm including just few optional comments.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Mastodon_5019.pdf](https://github.com/PipedreamHQ/pipedream/files/10323478/Mastodon_5019.pdf)

**Review by alysonturing — APPROVED** [review]
LGTM!

**Pull request merged**